crates/extension_host/src/wasm_host.rs

```diff
@@ -255,7 +383,7 @@ impl WasmHost {
 
             Ok(WasmExtension {
                 manifest: manifest.clone(),
-                work_dir: this.work_dir.clone().into(),
+                work_dir: this.work_dir.join(manifest.id.as_ref()).into(),
```

Comment:
Since this is a working directory for a given extension, it makes sense that it should already contain the extension's ID in the path.